### PR TITLE
Update authors

### PR DESCRIPTION
**Pull request by sylee957**
#### References to other Issues or PRs

Fixes #19906

#### Release Notes

NO ENTRY

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -810,21 +809,191 @@ jhanwar <f2015463@pilani.bits-pilani.ac.in>
...
 OrestisVaggelis <orestis22000@gmail.com>
 Nikhil Maan <nikhilmaan22@gmail.com>
 Abhinav Anand <abhinav.anand2807@gmail.com>
-sqs <googol.sqs@gmail.com>
+Qingsha Shi <googol.sqs@gmail.com>
 Juan Barbosa <js.barbosa10@uniandes.edu.co>
 Prionti Nasir <pdn3628@rit.edu>
+Bharat Raghunathan <bharatraghunthan9767@gmail.com>
+arooshiverma <av22@iitbbs.ac.in>
+Christoph Gohle <ctg@mpq.mpg.de>
```
Comment (oscarbenjamin) [inline AUTHORS:832]: Is this not Gohlke?

File: AUTHORS [hunk 2]
```diff
@@ -810,21 +809,191 @@ jhanwar <f2015463@pilani.bits-pilani.ac.in>
...
+KaustubhDamania <kaustubh.damania@gmail.com>
+w495 <w495@yandex-team.ru>
+Akhil Rajput <akh1lrjput@gmail.com>
+Akhil Rajput <43316490+akhil-dh1@users.noreply.github.com>
+Benjamin Wolba <mail@benjaminwolba.com>
+mcpl-sympy <59268464+mcpl-sympy@users.noreply.github.com>
+彭于斌 <1931127624@qq.com>
+Rudr Tiwari <rudrtiwari@gmail.com>
+Aaryan Dewan <aaryandewan@yahoo.com>
+Aaryan Dewan <49852384+aaryandewan@users.noreply.github.com>
```
Comment (oscarbenjamin) [inline AUTHORS:986]: I guess duplicates like this should be added to `.mailmap`

**Comment by oscarbenjamin** [comment]
Probably need to fix the .mailmap file first:
```console
$ bin/mailmap_update.py 
checking git authors...


Ambiguous email address error: each address should refer to a single
author. Disambiguate the following in .mailmap. Then re-run this
script.

	InCogNiTo124 <marijansmetko123@gmail.com>
	Marijan Smetko <marijansmetko123@gmail.com>

	Lagaras Stelios <stel.lag@hotmail.com>
	lagamura <stel.lag@hotmail.com>

	Milan Jolly <milan.cs16@iitp.ac.in>
	mijo2 <milan.cs16@iitp.ac.in>

	Naman Gera <namangera15@gmail.com>
	Namannimmo <namangera15@gmail.com>

	kunal <kunalgk1999@gmail.com>
	rationa-kunal <kunalgk1999@gmail.com>


Ambiguous name warning: if a person uses more than one email address,
entries should be added to .mailmap to merge them into a single
canonical address. Then re-run this script.

	Aaryan Dewan <49852384+aaryandewan@users.noreply.github.com>
	Aaryan Dewan <aaryandewan@yahoo.com>

	Akhil Rajput <43316490+akhil-dh1@users.noreply.github.com>
	Akhil Rajput <akh1lrjput@gmail.com>

	Naman Gera <43007653+namannimmo10@users.noreply.github.com>
	Naman Gera <namangera15@gmail.com>

	Peter Brady <petertbrady@gmail.com>
	Peter Brady <ptb@.lanl.gov>

	mohit <39158356+mohitacecode@users.noreply.github.com>
	mohit <42018918+mohitshah3111999@users.noreply.github.com>
```

**Comment by oscarbenjamin** [comment]
Was this generated using `bin/authors_update.py`?

File: bin/authors_update.py [hunk 3]
```diff
@@ -91,9 +91,9 @@ def move(l, i1, i2, who):
     assert 'Sergey B Kirpichev' == author_name(git_people.pop(226)
         ), 'Sergey B Kirpichev was not found at line 226.'
     assert 'azure-pipelines[bot]' == \
-        author_name(git_people.pop(751)), 'azure-pipelines[bot] was not found at line 751'
+        author_name(git_people.pop(750)), 'azure-pipelines[bot] was not found at line 750'
     assert 'whitesource-bolt-for-github[bot]' == \
-        author_name(git_people.pop(792)), 'whitesource-bolt-for-github[bot] not found at line 792'
+        author_name(git_people.pop(791)), 'whitesource-bolt-for-github[bot] not found at line 791'
```
Comment (oscarbenjamin) [inline bin/authors_update.py:96]: Hard-coding the indices doesn't seem like a great idea.

File: AUTHORS (hunk 1 above)
Comment (sylee957) [reply AUTHORS:832]: It was from the commits of #15885

**Comment by sylee957** [comment]
I see there are still some duplicate authors that wasn't automatically detected

**Comment by oscarbenjamin** [comment]
In the run up to 1.5 I emailed a bunch of people to ask what name they wanted to be recorded as and put all of those in the .mailmap. Looks like I missed the part of the release process that is updating the authors file.

**Comment by oscarbenjamin** [comment]
I think we should probably make it part of the process for merging a PR to sort these things out. There should be a test that all commits map (via the .mailmap) to an author listed in the authors file. If not then either the author needs to be added to the file or the .mailmap needs a new line for the alternate name/email address from the commits. The other possibility is that the author will want to fix their git config and update the commits to have their preferred name and email address.

I think it's better to do that at the time when merging a PR rather than later because it makes it clear to the author what is happening and also gives them the opportunity to specify what name they would like to be recorded as. In some cases authors did not want to be recorded under e.g. a particular email address but once the merge has happened there is no way to remove their email address from the commits.

**Comment by oscarbenjamin** [comment]
Looks good

File: .mailmap [hunk 4]
```diff
@@ -216,8 +216,9 @@ Duane Nykamp <nykamp@umn.edu>
 Mihai A. Ionescu <ionescu.a.mihai@gmail.com>
 Sarwar Chahal <chahal.sarwar98@gmail.com>
 Hamish Dickson <hamish.dickson@gmail.com>
-Peter Brady <petertbrady@gmail.com>
-Peter Brady <petertbrady@gmail.com> <ptb@lanl.gov>
+Peter Brady <petertbrady@gmail.com> Peter Brady <ptb@.lanl.gov>
+Peter Brady <petertbrady@gmail.com> Peter Brady <ptb@lanl.gov>
```
Comment (asmeurer) [inline .mailmap:220]: @pbrady should confirm which email he would prefer to have in the AUTHORS file.

Comment (pbrady) [reply .mailmap:220]: the gmail one is fine.

**Pull request merged**